### PR TITLE
Improve XML text extraction

### PR DESCRIPTION
**Pull request by pegiadise**
This PR:
- separates handling of text nodes and attributes in nodes
- adds support for ignoring text nodes in specific XML tags
- stops stringifying text between `<p>` tags

### CODE REVIEW

**Review by gartmeier — APPROVED** [review]

**Review by geokal94 — APPROVED** [review]

File: src/extract.js [hunk 1]
```diff
@@ -504,45 +505,28 @@ exports.Extractor = class Extractor {
...
-            if (matches) {
-              matches.forEach(match => {
-                const finalMatch = match.replace(/\\/, '');
-                textArray.push(finalMatch);
-              });
-            }
-          }
+      const element = node[0];
+      const text = element.data ? element.data.trim() : '';
+      const findTextsInExpression = (value) => value.match(new RegExp("(?<=').+?(?=')", 'g')) || [];
```
Comment (pegiadise) [inline src/extract.js:510]: @digi604 this is our only remaining problem right now because the `RegEx` is not 100% correct and the `.pot` output contains values such as
```
#: app.xml
msgid "+ DATE(currentUser.Last_Execution_Date__c) : "
msgstr ""

#: app.xml
msgid "+DATE(coaching.PCH__Execution_Date__c) :"
msgstr ""
```

Maybe it should be handled in a different PR? WDYT?
